webapp/src/js/services/contact-view-model-generator.js

```diff
@@ -211,6 +211,24 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
       });
     };
 
+    var updatePlaceName = function(reports) {
+      let promises = [];
+      reports.forEach(function(report) {
+        if (report.fields && report.fields.place_id) {
+          promises.push(
+            DB()
+              .get(report.fields.place_id)
+              .then(function(place) {
+                report.fields.place_name = place.name;
+              })
+          );
+        }
+      });
+
+      return $q.all(promises)
+               .then(() => reports);
```

Comment:
This isn't good for performance either because it's going to issue N queries to the database when we could just send 1 query for all the IDs.

However I think we should just do the same thing that the LHS of the Reports tab is doing rather then writing custom code. The LHS is generated by the LiveList service and if you look around line 112 you'll see a `getHeading` function which gets the subject name. This comes from the GetDataRecords service where if you request it without `include_docs:true` you get summaries from the GetSubjectSummaries service. If you can hook in to that and reuse that code that'd be ideal.

This is a complicated piece of code and it's also important we don't slow it down (we've just spent heaps of time trying to speed it up!) so it's worth taking the time to get this right.